### PR TITLE
PORT-6651 | Mini guide on connecting Jira issue to a service

### PR DESCRIPTION
**Pull request by lordsarcastic**
# Description
Added Mini guide on connecting Jira issue to a service

## Updated docs pages

Please also include the path for the updated docs

- Jira (`docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md`)

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 1]
```diff
@@ -1226,3 +1232,116 @@ The script requires the following environment variables:
...
+1. **Log in to Jira** as a user with the `Administer Jira` global permission;
+2. **Navigate to the issue** you want to label;
+3. **Click on the issue** to open it;
+4. **Click on the `Labels` field** on the right hand side to add a label;
+5. **Add a label** that represents the service the issue is related to, `port-auth-service`. For this guide, let's assume there is a service entity identified by `auth-service` in your `Service` blueprint in Port.
+
+<img src='/img/guides/jiraAddLabelToIssue.png' width='60%' border='1px' />
+
+:::note Control the tag name
+Since our Jira `issue` may already have several tags, we will need a mechanism to control how these tags will be related to our `Service` blueprint. A way to achieve this relation is to prefix the tag name with the keyword `port-`. We will then use JQ to select the tags that starts with this keyword. So, our example tag will be named `port-auth-service`, which will correspond to a Service entity identified by `auth-service` in Port.
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1262]: Replace all mention of tags to labels

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 2]
```diff
@@ -1226,3 +1232,116 @@ The script requires the following environment variables:
...
+:::
+
+### Create the service relation
+
+Now that Port is synced with our Jira resources, let's reflect the Jira issue in our services to display the issue associated with a service.
+First, we will need to create a [relation](/build-your-software-catalog/define-your-data-model/relate-blueprints/#what-is-a-relation) between our services and the corresponding Jira issue.
+
+1. Head back to the [Builder](https://app.getport.io/dev-portal/data-model), choose the `Jira issue` <PortTooltip id="blueprint">blueprint</PortTooltip>, and click on `New relation`:
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1270]: ... chose the Jira Issue (note the capitalisation of issue)

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 3]
```diff
@@ -1226,3 +1232,116 @@ The script requires the following environment variables:
...
+            parentIssue: .fields.parent.key
+            subtasks: .fields.subtasks | map(.key)
+            service: .fields.labels | map(select(startswith("port"))) | map(sub("port-"; ""; "g")) | .[0]
+```
+
+</details>
+
+:::tip JQ explanation
+
+The JQ below selects all labels that start with the keyword `port`. It then removes "port-" from each tag, leaving only the part that comes after it. It then selects the first match, which is equivalent to the service in Port.
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1327]: replace the mention of tag to label

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 4]
```diff
@@ -1226,3 +1232,116 @@ The script requires the following environment variables:
...
+```yaml
+service: .fields.labels | map(select(startswith("port"))) | map(sub("port-"; ""; "g")) | .[0]
+```
+
+:::
+
+What we just did was map the `Jira Issue` to the relation between it and our `Services`.  
+Now, if our `Service` identifier is equal to the Jira issue label, the `service` will automatically be linked to it &nbsp;🎉
+
+<img src='/img/guides/jiraIssueAfterServiceMapping.png' width='60%' border='1px' />
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1338]: This image is not rendering well. Please replace it

**Review by PeyGis — CHANGES_REQUESTED** [review]
Looks great. Just some minor comments

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 5]
```diff
@@ -1226,3 +1232,116 @@ The script requires the following environment variables:
...
+            priority: .fields.priority.id
+            created: .fields.created
+            updated: .fields.updated
+          relations:
+            board: .boardId | tostring
+            sprint: .sprint.id | tostring
+            project: .fields.project.key
+            parentIssue: .fields.parent.key
+            subtasks: .fields.subtasks | map(.key)
+            service: .fields.labels | map(select(startswith("port"))) | map(sub("port-"; ""; "g")) | .[0]
```
Comment (phalbert) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1320]: You could also highlight the modified line. It means a user can easily see the affected line. I eventually saw the tip below which would then explain why the change happened.

```javascript
// highlight-start
service: .fields.labels | map(select(startswith("port"))) | map(sub("port-"; ""; "g")) | .[0]
// highlight-end
```

File: docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md [hunk 6]
```diff
@@ -1503,3 +1505,118 @@ The script requires the following environment variables:
...
+<img src='/img/guides/jiraAddLabelToIssue.png' width='60%' border='1px' />
+
+:::note Control the label name
+Since our Jira `issue` may already have several labels, we will need a mechanism to control how these labels will be related to our `Service` blueprint. A way to achieve this relation is to prefix the label name with the keyword `port-`. We will then use JQ to select the labels that starts with this keyword. So, our example label will be named `port-auth-service`, which will correspond to a Service entity identified by `auth-service` in Port.
+:::
+
+### Create the service relation
+
+Now that Port is synced with our Jira resources, let's reflect the Jira Issue in our services to display the issue associated with a service.
+First, we will need to create a [relation](/build-your-software-catalog/define-your-data-model/relate-blueprints/#what-is-a-relation) between our services and the corresponding Jira Issue.
```
Comment (hadar-co) [inline docs/build-your-software-catalog/sync-data-to-catalog/jira/jira.md:1541]:
```suggestion
First, we will need to create a [relation](/build-your-software-catalog/customize-integrations/configure-data-model/relate-blueprints/) between our services and the corresponding Jira Issue.
```

**Review by hadar-co — APPROVED** [review]

**Pull request merged**